doc/nrf/gs_programming.rst

```diff
@@ -122,3 +122,92 @@ After completing the :ref:`manual <build_environment_cli>` or :ref:`automatic <g
       The ``west flash`` command automatically resets the kit and starts the application.
 
 For more information on building and programming using the command line, see the Zephyr documentation on :ref:`zephyr:west-build-flash-debug`.
+
+.. _gs_programming_output:
+
+Output build files
+******************
+
+The building process produces each time an *image file*.
+
+.. output_build_files_info_start
+
+The image file, also referred to as *executable*, *program*, or *elf file*, consists of pieces of code and data that are identified by image-unique names recorded in a single symbol table.
+The symbol table exists as metadata in a :file:`.elf` or :file:`.exe` file and is not included when the image is converted to a HEX file for programming.
+Instead, a linker places the code and data at their addresses.
+
+Depending on the application and the SoC, you can use one or several images.
+To determine if you have zero, one, or more images, count the number of times the linker runs.
```

Comment:
> count the number of times the linker runs.

How can I judge the number of images by the number of linker runs ?
Zephyr build system may create intermediate binaries, which means that when creating a final build, the linker may run 1, 2, or 3 times just for a single image.

Reply:
https://docs.zephyrproject.org/latest/build/cmake/index.html